presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -457,7 +457,7 @@ private void destroyIfNecessary()
         }
     }
 
-    private static ListenableFuture<?> isBlocked(Operator operator)
+    private static ListenableFuture<?> getBlockedFuture(Operator operator)
     {
         ListenableFuture<?> blocked = operator.isBlocked();
         if (blocked.isDone()) {
```

Comment:
Yea, but `Futures.allAsList` registers listeners in the source futures. Each time. Are you sure this won't create too many of them and the overall overhead in `Driver` won't be bad?